test/unit/org/apache/cassandra/schema/SchemaTest.java

```diff
@@ -78,4 +92,81 @@ public void testTransKsMigration() throws IOException
         }
     }
 
+    @Test
+    @BMRules(rules = { @BMRule(name = "client warning 1",
+                               targetClass = "CreateKeyspaceStatement",
+                               targetMethod = "apply",
+                               targetLocation = "AT INVOKE KeyspaceParams.validate",
+                               action = "org.apache.cassandra.schema.SchemaTest.addWarn()"),
+                       @BMRule(name = "client warning 2",
+                               targetClass = "CreateKeyspaceStatement",
+                               targetMethod = "clientWarnings",
+                               targetLocation = "AT EXIT",
+                               action = "return org.apache.cassandra.schema.SchemaTest.addWarnToList($!)"),
+                       @BMRule(name = "client warning 3",
+                               targetClass = "AlterSchemaStatement",
+                               targetMethod = "clientWarnings",
+                               targetLocation = "AT EXIT",
+                               action = "return org.apache.cassandra.schema.SchemaTest.addWarnToList($!)")})
+    public void testClientWarningsOnCreateKeyspace() throws Throwable
+    {
+        ClientWarn.instance.captureWarnings();
+        injectedWarnings.clear();
+        createKeyspace("CREATE KEYSPACE %s WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}");
+
+        Assertions.assertThat(injectedWarnings).contains(msg1, msg2); // failure here means the bm rules need to be updated
+        Assertions.assertThat(ClientWarn.instance.getWarnings()).containsExactlyInAnyOrder(msg1, msg2);
+    }
+
+    @Test
+    @BMRules(rules = { @BMRule(name = "client warning 1",
+                               targetClass = "CreateTableStatement",
+                               targetMethod = "apply",
+                               targetLocation = "AT INVOKE TableMetadata.validate",
+                               action = "org.apache.cassandra.schema.SchemaTest.addWarn()"),
+                       @BMRule(name = "client warning 2",
+                               targetClass = "CreateTableStatement",
+                               targetMethod = "clientWarnings",
+                               targetLocation = "AT EXIT",
+                               action = "return org.apache.cassandra.schema.SchemaTest.addWarnToList($!)"),
+                       @BMRule(name = "client warning 3",
+                               targetClass = "AlterSchemaStatement",
+                               targetMethod = "clientWarnings",
+                               targetLocation = "AT EXIT",
+                               action = "return org.apache.cassandra.schema.SchemaTest.addWarnToList($!)"),
```

Comment:
missing case for `alter`

Reply:
yup, I can added alter as well, but I don't think it makes much sense - this way we should add for each single schema alteration statement. I added "create table" because for this case the code was actually failing the tests - if the warning was added inside `apply`, it was not returned to the client (it was because the workaround was applied only to create and alter keyspace)

Reply:
given this patch/tests are showing that executors are doing the right thing; I am cool not testing every DDL, but if its dead code then should remove

Reply:
yes, I'll remove it, just was off for the last two days